soufi/tests/finders/test_centos_finder.py

```diff
@@ -38,289 +44,238 @@ def make_top_page_content(self, versions):
+        self.assertEqual([url], disc_source.urls)
+
+    def test_initializer_finds_repos_when_called_with_no_args(self):
+        name = self.factory.make_string('name')
+        version = self.factory.make_string('version')
+        get_source_repos = self.patch(centos.CentosFinder, '_get_source_repos')
+        get_binary_repos = self.patch(centos.CentosFinder, '_get_binary_repos')
+        centos.CentosFinder(name, version, SourceType.os)
+        get_source_repos.assert_called_once_with(centos.DEFAULT_SEARCH, False)
+        get_binary_repos.assert_called_once_with(centos.DEFAULT_SEARCH, False)
 
     def test__get_dirs(self):
         finder = self.make_finder()
-        versions = ['4.5', '7.0', '7.1', '8.0']
-        versions_ = versions + ['ignored']
-        data = self.make_top_page_content(versions_)
-        get = self.patch(requests, 'get')
-        get.return_value = self.make_response(data, requests.codes.ok)
-        dirs = finder._get_dirs()
-        self.assertEqual(versions, dirs)
-        get.assert_called_once_with(
-            "https://vault.centos.org/centos/", timeout=30
-        )
-
-    def test__get_path(self):
-        finder = self.make_finder()
-        data = self.make_index_page_content(finder.name, finder.version)
-        ref = finder._get_path(data)
-        self.assertEqual(f'{finder.name}-{finder.version}.src.rpm', ref)
-
-    def test__get_path_returns_None_if_not_found(self):
-        finder = self.make_finder()
-        data = self.make_index_page_content(
-            self.factory.make_string(), self.factory.make_string()
-        )
+        top_repos = ('1.0.123', '2.1.3456', 'bogus', '3.7.89', '3')
+        top_data = self.make_top_page_content(top_repos)
         get = self.patch(requests, 'get')
-        get.return_value = self.make_response(data, requests.codes.ok)
-        dir_ = self.factory.make_string('8.0') + '/'
-        self.assertIsNone(finder._get_path(dir_))
+        get.return_value = self.make_response(top_data, requests.codes.ok)
+        result = list(finder._get_dirs())
+        # Ensure that only the items we're interested in come back
+        self.assertEqual([top_repos[3], top_repos[1], top_repos[0]], result)
+        get.assert_called_once_with(centos.VAULT, timeout=30)
 
-    def test_find_raises_when_not_found(self):
+    def test__get_source_repos(self):
         finder = self.make_finder()
-        url = self.factory.make_url()
-        self.patch(finder, '_get_dirs').return_value = ['a']
-        self.patch(finder, '_get_source_dirs_content').return_value = [
-            (url, 'data')
+        dirs = [self.factory.make_string() for _ in range(3)]
+        subdirs = [self.factory.make_string() for _ in range(3)]
+        get_dirs = self.patch(finder, '_get_dirs')
+        get_dirs.return_value = dirs
+        test_url = self.patch(finder, '_test_url')
+        test_url.return_value = True
+        result = list(finder._get_source_repos(subdirs, False))
```

Comment:
OK, so this one is REALLY esoteric, but:

I am still dealing with this regression: https://youtrack.jetbrains.com/issue/PY-48718

...which means my test runner swallows most output errors and simply prints "OK" and "Failure" when running.  So it really helps me to bind damn-near everything in tests to an lvalue so I can set a breakpoint and look at the test results in a debugger when tests fail.  So while I can write comparisons with `testtools.matchers.SameMembers`, and it will just use the generator and Do The Right Thing, this test pattern is far more useful to me.

Reply:
Ok no worries. I hate tools sometimes.

> It's been over 6 months, and I ostensibly pay for support.

Ouch :laughing: